pkg/gadgets/snapshot/process/tracer/tracer.go

```diff
@@ -120,3 +143,88 @@ func RunCollector(config *Config, enricher gadgets.DataEnricher) ([]*processcoll
 
 	return events, nil
 }
+
+func getPidEvents(config *Config, enricher gadgets.DataEnricher, pid int) ([]*processcollectortypes.Event, error) {
+	var events []*processcollectortypes.Event
+	var val uint32
+
+	items, err := os.ReadDir(fmt.Sprintf("/proc/%d/task/", pid))
+	if err != nil {
+		return nil, err
+	}
+
+	for _, item := range items {
+		if !item.IsDir() {
+			continue
+		}
+
+		tid64, err := strconv.ParseUint(item.Name(), 10, 32)
+		if err != nil {
+			continue
+		}
+		tid := int(tid64)
+
+		commBytes, _ := ioutil.ReadFile(fmt.Sprintf("/proc/%d/comm", tid))
+		comm := strings.TrimRight(string(commBytes), "\n")
+		mntnsid, _ := containerutils.GetMntNs(tid)
+
+		if config.MountnsMap != nil {
+			// TODO: This would be more efficient to store
+			// these elements in user space to avoid
+			// performing systemcalls to lookup in the eBPF
+			// map
+			err := config.MountnsMap.Lookup(&mntnsid, &val)
+			if err != nil {
+				continue
+			}
+		}
+
+		event := processcollectortypes.Event{
+			Event: eventtypes.Event{
+				Type: eventtypes.NORMAL,
+			},
+			Tid:       tid,
+			Pid:       pid,
+			Command:   comm,
+			MountNsID: mntnsid,
+		}
+
+		if enricher != nil {
+			enricher.Enrich(&event.CommonData, event.MountNsID)
+		}
+
+		events = append(events, &event)
+	}
+
+	return events, nil
+}
+
+func runProcfsCollector(config *Config, enricher gadgets.DataEnricher) ([]*processcollectortypes.Event, error) {
+	items, err := os.ReadDir("/proc/")
+	if err != nil {
+		return nil, err
+	}
+
+	var events []*processcollectortypes.Event
+
+	for _, item := range items {
+		if !item.IsDir() {
+			continue
+		}
+
+		pid64, err := strconv.ParseUint(item.Name(), 10, 32)
+		if err != nil {
+			continue
+		}
+		pid := int(pid64)
+
+		pidEvents, err := getPidEvents(config, enricher, pid)
+		if err != nil {
+			return nil, err
```

Comment:
It should `continue` and silently ignore the error if it cannot read a file "/proc/%d/*": the process might have terminated between the `os.ReadDir("/proc/")` and the subsequent reads.